cadl-extension/src/code-model-builder.ts

```diff
@@ -133,6 +135,7 @@ export class CodeModelBuilder {
   public constructor(program1: Program, options: EmitterOptions) {
     this.options = options;
     this.program = program1;
+    this.dpgContext = { program: this.program, generateProtocolMethods: true, generateConvenienceMethods: true };
```

Comment:
Does not need to be in this PR, but cadl-dpg might want us to read the 2 configures from emitter options. Please double check with Pan.

Reply:
Sure, I will double check with Pan, reading the configurations from emitter makes more sense. 

PS: I checked with Pan before, cadl-dpg has [createDpgContext](https://github.com/Azure/cadl-azure/blob/main/packages/cadl-dpg/src/decorators.ts#L250) API,  it takes `EmitContext<DpgEmitterOptions>`, but we don't pass in `EmitContext` to code-model-builder.ts, we need to update the constructor to pass in EmitContext if we want to use `createDpgContext`. Currently for conveniency, I just create DpgContext directly. Please let me know your thoughts on it.

Reply:
Yes, you can pass `context: EmitContext<EmitterOptions>` from emitter to ctor of CodeModelBuilder (replace that of `options: EmitterOptions` which is just `context.options`.